modernizer-maven-plugin/src/test/java/org/gaul/modernizer_maven_plugin/ModernizerSuppressionsTest.java

```diff
@@ -0,0 +1,287 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.modernizer_maven_plugin;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClassConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningDeclaredTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningGenericTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithArrayTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithDeclaredTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithEmptyParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveAndGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithVoidParameterTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreStaticOuterClassConstructor.IgnoreStaticInnerClassConstructor;
+import org.junit.Before;
+import org.junit.Test;
+import org.objectweb.asm.ClassReader;
+
+public final class ModernizerSuppressionsTest {
```

Comment:
Maybe add something like `EndToEnd` to the name of the class so it's clear that it's testing **everything**

Reply:
Will do this before merging to prevent messing the diff.

Reply:
The whole class is new in this PR, so you can do it whenever. The renames mess up the diff when the class existed before the PR

Reply:
Oh right. Will change it.